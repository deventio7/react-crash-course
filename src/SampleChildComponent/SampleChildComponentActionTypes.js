const SAMPLE_CHILD_COMPONENT_SUBMIT = 'SAMPLE_CHILD_COMPONENT_SUBMIT';

export default {
  SAMPLE_CHILD_COMPONENT_SUBMIT
};