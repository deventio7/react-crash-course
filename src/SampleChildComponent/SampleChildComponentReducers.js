import SampleChildComponentActionTypes from './SampleChildComponentActionTypes';

const initialState = {
	storedText: ''
}

const sampleChildSubmit = (state = initialState, action) => {
	switch (action.type) {
		case SampleChildComponentActionTypes.SAMPLE_CHILD_COMPONENT_SUBMIT:
			return {
				...state,
				storedText: action.payload
			};
		default:
			return state;
	}
}

export default {
  sampleChildSubmit
};