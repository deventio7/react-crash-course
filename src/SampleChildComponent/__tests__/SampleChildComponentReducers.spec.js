import SampleChildComponentActionTypes from '../SampleChildComponentActionTypes';
import SampleChildComponentReducers from '../SampleChildComponentReducers';

describe('SampleChildComponentReducers', () => {
  it('should return initial state', () => {
    const otherAction = {
      type: 'test'
    };
    const result = SampleChildComponentReducers.sampleChildSubmit(undefined, otherAction);
    expect(result).toMatchSnapshot();
  });

  it('should change state appropriately when receiving SAMPLE_CHILD_COMPONENT_SUBMIT', () => {
    const createAlertAction = {
      type: SampleChildComponentActionTypes.SAMPLE_CHILD_COMPONENT_SUBMIT,
      payload: 'TEST TEXT'
    };
    const result = SampleChildComponentReducers.sampleChildSubmit(undefined, createAlertAction);
    expect(result).toMatchSnapshot();
  });
});