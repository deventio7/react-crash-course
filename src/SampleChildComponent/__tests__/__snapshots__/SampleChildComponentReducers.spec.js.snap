// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleChildComponentReducers should change state appropriately when receiving SAMPLE_CHILD_COMPONENT_SUBMIT 1`] = `
Object {
  "storedText": "TEST TEXT",
}
`;

exports[`SampleChildComponentReducers should return initial state 1`] = `
Object {
  "storedText": "",
}
`;
